themes/bootstrap5/templates/turnstile/challenge.phtml

```diff
@@ -0,0 +1,22 @@
+<?php
+  $this->headScript()->appendFile('https://challenges.cloudflare.com/turnstile/v0/api.js', 'text/javascript', ['defer' => '', 'async' => '']);
+  $js = <<<JS
+      function turnstileChallengeEnded(token) {
+          document.getElementById('turnstile_token').setAttribute('value', token);
+          document.getElementById('turnstile_form').submit();
+      }
+      JS;
+?>
+<?=$this->inlineScript()->appendScript($js)?>
+
+<div
+    class="cf-turnstile"
+    data-sitekey="<?=$this->escapeHtml($siteKey)?>"
+    data-callback="turnstileChallengeEnded"
+    data-error-callback="turnstileChallengeEnded"
+></div>
+<form id="turnstile_form" method="post" action="/vufind/Turnstile/Verify">
+    <input id="turnstile_token" name="token" type="hidden">
+    <input name="policyId" type="hidden" value="<?=$this->escapeHtml($policyId)?>">
+    <input name="destination" type="hidden" value="<?=$this->escapeHtml($destination)?>">
```

Comment:
I'm escaping these two variables (and $siteKey above) even though they never *should* come from user input, but they flowed through URL params on the way here, so it seemed the safe thing to do.  If someone actually does mess with the params intentionally
- a fake $siteKey would in theory let the attacker supply their own key, but in Turnstile configuration their key would not be set to work with this domain, so the challenge should not return a valid token.
- a fake $policyId could let them substitute a more lenient policy, I guess, that they happen to know is also configured on the same server?  That one I guess is a risk
- a fake $destination ... would just take them to a fake destination after success, but that destination is still subject to the rate limiter, so no point there.

Reply:
Yeah, I would welcome some insights on that, not sure how to make it work.  I guess the salt would be a random value secret in the config file, that the institution would have to create?

Reply:
There's a Security/HMACkey setting in config.ini that might be suitable for salting purposes. (It also looks like it might be valuable to randomize this as part of the InstallController, but that's a separate project).

Reply:
I'm using HMAC now.